src/Utils.ts

```diff
@@ -16,17 +16,24 @@ export function fromEditorConfig(
 ): TextEditorOptions {
 	const resolved: TextEditorOptions = {
 		tabSize: (config.indent_style === 'tab'
+			|| config.indent_size === 'tab'
 			? get(config, 'tab_width', config.indent_size)
-			: get(config, 'indent_size', config.tab_width)
-		)
+			: config.indent_style === 'space'
+			? get(config, 'indent_size', config.tab_width)
+			: get(config, 'indent_size', config.tab_width, defaults.tabSize)
+		),
+		insertSpaces: config.indent_style
+			? config.indent_style === 'space'
+			: defaults.insertSpaces
 	};
-	if (get(resolved, 'tabSize') === 'tab') {
-		resolved.tabSize = config.tab_width;
+	if ((!config.indent_style || config.indent_style === 'unset')
+		&& config.indent_size !== 'tab'
+		&& defaults && Object.keys(defaults).length === 0) {
+			resolved.tabSize = 'auto';
+			resolved.insertSpaces = 'auto';
```

Comment:
Similarly, this is also overridden by `editor.detectIndentation`, but the default, built-in editor value is `true`.
```json
  // Insert spaces when pressing `Tab`.
  // This setting is overridden based on the file contents
  // when `editor.detectIndentation` is on.
  "editor.insertSpaces": true,
```